### PR TITLE
Use Segment and GA to track request loan actions

### PR DESCRIPTION
**Pull request by chrismin**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -36,5 +36,62 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
+
+    <script>
```
Comment (saturnial) [inline public/index.html]: is it possible to house this in a separate js file and reference it. It's pretty noisy. I would also reference where you pulled it from so future devs don't think we wrote it.

File: src/utils/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export class Analytics {
+    static page(pageName: string, properties?: any) {
```
Comment (saturnial) [inline src/utils/analytics.ts]: properties is an `object`, no?

File: src/utils/analytics.ts [hunk 3]
```diff
@@ -1,17 +1,37 @@
...
+    export enum RequestLoanAction {
+        ViewForm = "View Form",
+        BeginForm = "Begin Form",
+        SubmitForm = "Submit Form",
+        ConfirmRequest = "Confirm Request",
+        ViewConfirmation = "View Confirmation",
+    }
+
+    // The union of all Action enums
+    type Action = RequestLoanAction;
```
Comment (saturnial) [inline src/utils/analytics.ts]: this isn't a union. but a typealias. did you mean to make it a union?

Comment (chrismin) [reply src/utils/analytics.ts]: I meant for it to be a placeholder of sorts... right now, there only Action enum is `RequestLoanAction`, but in the future we'll probably have `FillLoanAction`, etc... but it felt weird to have the parameter type below be `RequestLoanAction` and not the more generic `Action`.

So, in a future PR, we'll probably see

`type Action = RequestLoanAction | FillLoanAction;`

Comment (saturnial) [reply src/utils/analytics.ts]: Makes sense.

File: src/utils/analytics.ts (hunk 2 above)
Comment (chrismin) [reply src/utils/analytics.ts]: Done.

**Pull request merged**